pre_commit/git.py

```diff
@@ -159,7 +166,8 @@ def git_path(name, repo='.'):
 
 def check_for_cygwin_mismatch():
     """See https://github.com/pre-commit/pre-commit/issues/354"""
-    if sys.platform in ('cygwin', 'win32'):  # pragma: no cover (windows)
+    if sys.platform in ('cygwin', 'win32') and \
+            not is_mingw():  # pragma: no cover (windows)
```

Comment:
hmmm I think instead this needs updating for four-way checking now since any of the following mismatches are problematic:

- windows
- cygwin
- msys
- mingw

Reply:
From your Python overview, this shouldn't happen, as in: this check should be sufficient? As far as I know, only mingwx (under MSYS2) 'is troublesome'

Reply:
the check here is to ensure that `cygpath` is never needed.  If `cygpath` is needed there's a mismatch between the flavor of `git` and `python` which is not supported